packages/pigment-css-react/src/styled.js

```diff
@@ -112,40 +112,31 @@ export default function styled(tag, componentMeta = {}) {
         });
       }
 
-      const finalClassName = clsx(classes, sxClass, className, getVariantClasses(props, variants));
-      const toPassProps = Object.keys(restProps)
-        .filter((item) => {
-          const res = shouldForwardProp(item);
-          if (res) {
-            return defaultShouldForwardProp(item);
-          }
-          return false;
-        })
-        .reduce((acc, key) => {
-          acc[key] = restProps[key];
-          return acc;
-        }, {});
+      const finalClassName = clsx(
+        classes,
+        sxClass,
+        className,
+        getVariantClasses(inProps, variants),
+      );
 
-      // eslint-disable-next-line no-underscore-dangle
-      if (!Component.__isStyled || typeof Component === 'string') {
-        return (
-          // eslint-disable-next-line react/jsx-filename-extension
-          <Component
-            {...toPassProps}
-            ref={ref}
-            className={finalClassName}
-            style={{
-              ...varStyles,
-              ...style,
-            }}
-          />
-        );
+      const newProps = {};
+      // eslint-disable-next-line no-restricted-syntax
+      for (const key in props) {
```

Comment:
I hope this is comparatively faster than `Object.keys` since optimizing this `styled` call is the most important part as it'll be called a lot.

Reply:
I think so, it's the same logic as emotion.

Reply:
styled-component also [do the same](https://github.com/styled-components/styled-components/blob/22e8b7f233e12500a68be4268b1d79c5d7f2a661/packages/styled-components/src/models/StyledComponent.ts#L142).